lib/types/record.nix

```diff
@@ -0,0 +1,116 @@
+{ lib }:
+
+let
+  inherit (lib)
+    mapAttrs
+    concatMapAttrs
+    zipAttrs
+    removeAttrs
+    attrNames
+    showOption
+    optional
+    optionalAttrs
+    mergeDefinitions
+    mkOptionType
+    isAttrs
+    ;
+
+  inherit (lib.options)
+    showDefs
+    ;
+
+  inherit (lib.strings)
+    escapeNixIdentifier
+    ;
+
+  record =
+    {
+      fields ? { },
+      freeformType ? null,
```

Comment:
The freeform type applies at the root of a submodule / configuration, whereas this one applies to individual field values; not the attribute set containing them. I think that's a significant difference that should be reflected in the name. I think it was `wildcardType`.